src/Compilers/Core/Portable/FileSystem/PathUtilities.cs

```diff
@@ -107,6 +107,25 @@ internal static string GetDirectoryName(string path, bool isUnixLike)
             return null;
         }
 
+        internal static bool IsDirectoryChildOf(string child, string parent)
+        {
+            parent = RemoveTrailingDirectorySeparator(parent);
+
+            while (child != null)
+            {
+                child = RemoveTrailingDirectorySeparator(child);
+
+                if (child.Equals(parent, StringComparison.OrdinalIgnoreCase))
+                {
+                    return true;
```

Comment:
>return true; [](start = 20, length = 12)

It looks like we will get here if the function is called with equal arguments. Is this what we want?